frontend/components/feature/query-filters/multiselect-filter.tsx

```diff
@@ -76,26 +79,85 @@ export function MultiSelectFilter({ field, maxVisibleItems = 5 }: IProps) {
     appId,
   );
 
+  // Initialize selectedItems with query builder + global sessionStorage
+  const getInitialSelectedItems = (): string[] => {
+    // Try sessionStorage first
+    try {
+      const stored = sessionStorage.getItem(globalStorageKey);
+      if (stored) {
+        const allTempSelections = JSON.parse(stored);
+        const fieldSelections = allTempSelections[field.key];
+        if (fieldSelections) {
+          return fieldSelections;
+        }
+      }
+    } catch (error) {
+      // Error reading global storage - continue with query builder values
```

Comment:
Should we do something with these errors? (same goes for other empty `catch` in the component)